src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+
+export const PeoplePage: React.FC = () => {
+  const [peopleList, setPeopleList] = useState<Person[]>([]);
+  const [isShowError, setIsShowError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const { personSlug = '' } = useParams();
+
+  function findParent(name: string) {
+    return peopleList.find(person => person.name === name);
+  }
+
+  function getParent() {
+    return peopleList.map(person => ({
+      ...person,
+      fatherName: person.fatherName ? person.fatherName : '-',
+      motherName: person.motherName ? person.motherName : '-',
+      father: person.fatherName ? findParent(person.fatherName) : null,
+      mother: person.motherName ? findParent(person.motherName) : null,
+    }));
+  }
+
+  async function loadingPeople() {
+    setIsLoading(true);
+    const peopleFromServer = await getPeople();
+
+    try {
+      setPeopleList(peopleFromServer);
+      setIsShowError(false);
+    } catch {
+      setIsShowError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  useEffect(() => {
+    loadingPeople();
+  }, []);
+
+  const people = useMemo(() => {
+    return getParent();
+  }, [peopleList]);
```

Comment:
You need to pass getParent function to the dependency array and wrap it into useCallback. It'll be the correct way to memorize this value.
```suggestion
  const people = useMemo(() => {
    return getParent();
  }, [getParent]);
```